test/Spec/MockContract.hs

```diff
@@ -159,12 +198,75 @@ runPABEffectPure config initState req =
     go (UploadDir dir) = mockUploadDir dir
     go (QueryChainIndex query) = mockQueryChainIndex query
 
-mockCallCommand :: ((Text, [Text]) -> MockContract String) -> ShellArgs a -> MockContract a
-mockCallCommand handleCliCommand ShellArgs {cmdName, cmdArgs, cmdOutParser} = do
+mockCallCommand ::
+  forall (a :: Type).
+  ShellArgs a ->
+  MockContract a
+mockCallCommand ShellArgs {cmdName, cmdArgs, cmdOutParser} = do
   tell $ map Text.unpack cmdArgs
   modify (\st -> st {commandHistory = cmdName <> " " <> Text.unwords cmdArgs : st.commandHistory})
 
-  cmdOutParser <$> handleCliCommand (cmdName, cmdArgs)
+  case (cmdName, cmdArgs) of
+    ("cardano-cli", "query" : "utxo" : "--address" : addr : _) ->
+      cmdOutParser <$> mockQueryUtxo addr
+    ("cardano-cli", "transaction" : "build" : _) ->
+      pure $ cmdOutParser ""
+    ("cardano-cli", "transaction" : "sign" : _) ->
+      pure $ cmdOutParser ""
+    ("cardano-cli", "transaction" : "submit" : _) ->
+      pure $ cmdOutParser ""
+    _ -> throwError @Text "Unknown command"
+
+mockQueryUtxo :: Text -> MockContract String
+mockQueryUtxo addr = do
+  state <- get @MockContractState
+
+  let network = state.contractEnv.cePABConfig.pcNetwork
+  pure $
+    mockQueryUtxoOut $
+      filter
+        ((==) addr . unsafeSerialiseAddress network . Ledger.txOutAddress . snd)
+        state.utxos
+
+mockQueryUtxoOut :: [(TxOutRef, TxOut)] -> String
+mockQueryUtxoOut utxos =
```

Comment:
This function has *quite* a lot of scope, could we split it up to be a little less massive?

Reply:
Absolutely